### DISCUSSION TITLE
pragma cannot be set when runtime is automatic

### DISCUSSION THREAD

**Discussion by mwmwmw** [algolia/react-recommendations#166] [Q&A] [answered]
In NextJS 14.0.4 I have a pretty basic related posts recommender. 

On Next 14.0.1 I get no error, and when I upgrade to 14.0.4 I get this error. 

I would log an issue, but I'm not sure this is an algolia problem. It doesn't like the `/** @jsx createElement */`

Has anyone encountered an issue like this before?

```
 ⨯ ./node_modules/.pnpm/@algolia+recommend-vdom@1.11.0/node_modules/@algolia/recommend-vdom/dist/esm/FrequentlyBoughtTogether.js
Error: 
  × pragma cannot be set when runtime is automatic
   ╭─[/home/erebus/rdar/node_modules/.pnpm/@algolia+recommend-vdom@1.11.0/node_modules/@algolia/recommend-vdom/dist/esm/FrequentlyBoughtTogether.js:1:1]
 1 │ import _extends from "@babel/runtime/helpers/extends";
 2 │ /** @jsx createElement */
   · ─────────────────────────
 3 │ import { createDefaultChildrenComponent } from './DefaultChildren';
 4 │ import { createDefaultFallbackComponent } from './DefaultFallback';
 5 │ import { createDefaultHeaderComponent } from './DefaultHeader';
 ```

**Comment by mwmwmw** [comment]
If I find/replace all `/** @jsx createElement */` with `/** @jsxRuntime classic */` everything works.

Obviously, that's not a good solution but demonstrates that all of the code is fine. If you know where I can configure this, I have already checked the next compiler docs that I believe were relevant https://nextjs.org/docs/architecture/nextjs-compiler.

**Comment by willpiers** [comment]
@mwmwmw Have you learned anything more about this issue in the last few weeks? I've had the same problem, but have been putting off dealing with it. Is there something we can change in our `next.config.js` files that will resolve this issue?

**Reply by mwmwmw** [reply]
Same, I was coming here to ask this question and realized I had already asked! 

I'm trying to upgrade a project to 14.1.0. 

There must be some setting, but I can't figure out what it is. 

All deps are up to date. 
I have tried node 20 and 21.
npm, pnpm, yarn all the same result

**Comment by raed667** [comment, marked as answer]
Should be fixed with the latest release: https://github.com/algolia/recommend/releases/tag/v1.13.0
